zebra-chain/src/types/amount.rs

```diff
@@ -0,0 +1,117 @@
+//!
+use std::{
+    convert::{TryFrom, TryInto},
+    marker::PhantomData,
+    ops::Range,
+};
+
+/// A runtime validated type for representing amounts of zatoshis
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Serialize, Deserialize)]
+pub struct Amount<C = NegativeAllowed>(i64, PhantomData<C>);
+
+impl<C> From<Amount<C>> for i64 {
+    fn from(amount: Amount<C>) -> Self {
+        amount.0
+    }
+}
+
+impl<C> From<Amount<C>> for u64 {
+    fn from(amount: Amount<C>) -> Self {
+        amount.0 as _
```

Comment:
If `C` is a marker type representing the bounds on `amount.0`, but we don't have any bounds on `C`, then we don't have any bounds on `amount.0`, right? In that case, how can we safely as-cast to `u64`?

Should this be `impl From<Amount<NonNegative>> for u64` or something similar?

Reply:
yes, good catch